packages/website/pages/docs/how-to/store-directory.md

```diff
@@ -42,14 +42,18 @@ async function main() {
     console.error(`usage: ${process.argv[0]} ${process.argv[1]} <directory-path>`)
   }
   const directoryPath = process.argv[2]
-  const files = await getFilesFromPath(path, {
+  const files = (async function* () {
+    for await (const file of filesFromPath(directoryPath, {
       pathPrefix: path.resolve(directoryPath), // see the note about pathPrefix below
-      hidden: true // use the default of false if you want to ignore files that start with '.'
-  })
+      hidden: true, // use the default of false if you want to ignore files that start with '.'
+    })) {
+      yield new File([file.stream()], file.name);
```

Comment:
constructing this explicitly isn't required for the js to work, but it is required for ts auto inference to work.

If we want to simplify this further, we could update `NFTStorage.storeDirectory` to accept `IterableOrAsyncIterable<FileObject|File>` (not just `File` but `FileObject` too).

It can also look similar by just not caring about ts auto inference, but IMO it's nice to be ready for a dev to not want to have to manually grok type errors

Reply:
If `storeDirectory` accepted `FileObject`, can we just feed it the output of `filesFromPath` directly? If so, that seems worth it. I've been confused by this same type error before, and it would be nice to sort it out for good.

Reply:
Yes

Reply:
@yusefnapora implemented in https://github.com/nftstorage/nft.storage/pull/1924